### PR TITLE
DOC Added the colormap references back

### PR DESCRIPTION
**Pull request by NelleV**
For fast access. The user should not have to go to the tutorials to get the list of available colormaps.

### CODE REVIEW

**Comment by anntzer** [comment]
I don't have a strong opinion as to whether to include it in the examples as well (I'm not sure it's more discoverable than the tutorials, but why not).  However it would definitely be useful to add a link from http://matplotlib.org/api/cm_api.html to the relevant examples/tutorials.

**Comment by choldgraf** [comment]
+1 on adding to examples, I think it's a more quickly-discoverable place for this kind of info. Also +1 on generally cross-linking as much as possible

File: examples/color/colormap_reference.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+Qualitative:
+    These colormaps vary rapidly in color. Qualitative colormaps are useful
+for
+    choosing a set of discrete colors. For example::
+
+        color_list = plt.cm.Set3(np.linspace(0, 1, 12))
+
+    gives a list of RGB colors that are good for plotting a series of lines on
+    a dark background.
+
```
Comment (choldgraf) [inline examples/color/colormap_reference.py]: Is it worth adding a section (maybe in another PR?) that lists all colorblind-friendly colormaps?

Comment (story645) [reply examples/color/colormap_reference.py]: I'm all for that/think that's a great idea.

Comment (NelleV) [reply examples/color/colormap_reference.py]: I don't think this should be added to this example. This is a reference list. This should be (if it isn't already) in the colormap tutorial.

Comment (story645) [reply examples/color/colormap_reference.py]: Why shouldn't a reference list provide accessibility references?

Comment (efiring) [reply examples/color/colormap_reference.py]: Can this be handled by including a reference here to the tutorial section?

**Comment by NelleV** [comment]
I didn't write it. This is just the old example (which is by the way still hanging around in the website, from an old build, and links misleading to some old hidden references and documentation).

The text and documentation can definitely be improved, but I don't have time to do this.

File: examples/color/colormap_reference.py (hunk 1 above)
Comment (story645) [reply examples/color/colormap_reference.py]: I think so, so long as there's an explicit mention of colorblind maps, something "See the tutorial for a more detailed explanation of colormaps, including information on colorblind friendly"

Comment (anntzer) [reply examples/color/colormap_reference.py]: I would just delete *all* the text (it's a reference!) and replace it by "for an in-depth overview of colomaps, including discussion about colorblind-friendliness, see (tutorial)".

**Comment by anntzer** [comment]
I think it's clearly an improvement, further fixes will go to a separate PR.

File: examples/color/colormap_reference.py (hunk 1 above)
Comment (NelleV) [reply examples/color/colormap_reference.py]: There is no information on colorblind maps in the tutorial, just outside references.

Feel free to merge or close if you feel this is not an improvement to the documentation.

**Comment by tacaswell** [comment]
@meeksdev backport to v2.1.0-doc

**Pull request merged**